pkg/api/data.go

```diff
@@ -0,0 +1,152 @@
+package api
+
+import (
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/tsdb"
+)
+
+// QueryData endpoint for querying backend data source plugins.
+// POST /api/ds/query
+func (hs *HTTPServer) QueryData(c *models.ReqContext, reqDto dtos.QueryDataRequest) Response {
+	if len(reqDto.Queries) == 0 {
+		return Error(500, "No queries found in query", nil)
```

Comment:
Isn't this wording more precise?
```suggestion
		return Error(500, "No queries found in request", nil)
```